tony-history-server/app/utils/HdfsUtils.java

```diff
@@ -0,0 +1,97 @@
+package utils;
+
+import java.io.BufferedReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import play.Logger;
+
+
+/**
+ * The class handles all HDFS file operations
+ */
+public class HdfsUtils {
+  private static final Logger.ALogger LOG = Logger.of(HdfsUtils.class);
+
+  static boolean isPathValid(FileSystem fs, Path filePath) {
+    try {
+      return fs.exists(filePath);
+    } catch (IOException e) {
+      LOG.error(filePath.toString() + " doesn't exist!");
+      return false;
+    }
+  }
+
+  static String contentOfHdfsFile(FileSystem fs, Path filePath) {
+    StringBuilder fileContent = new StringBuilder();
+    try (FSDataInputStream inStrm = fs.open(filePath);
+        BufferedReader bufReader = new BufferedReader(new InputStreamReader(inStrm))) {
+      String line;
+      while ((line = bufReader.readLine()) != null) {
+        fileContent.append(line);
+      }
+      return fileContent.toString();
+    } catch (IOException e) {
+      LOG.error("Couldn't read content of file from HDFS", e);
+      return "";
+    }
+  }
+
+  static List<Path> getValidPaths(FileStatus[] lsJobDir, Predicate<FileStatus> fn) {
+    return Arrays.stream(lsJobDir)
+        .filter(fn)
+        .map(FileStatus::getPath)
+        .collect(Collectors.toList());
+  }
+
+  public static List<Path> getFilePathsFromAllJobs(FileSystem fs, String histFolder, String fileType) {
+    List<Path> paths = new ArrayList<>();
+    FileStatus[] lsHist;
+    try {
+      lsHist = fs.listStatus(new Path(histFolder));
+    } catch (Exception e) {
```

Comment:
Please catch the specific exceptions thrown by listStatus() rather than doing a blanket catch of all Exception types. In general, it's best practice to catch as specific Exceptions as possible.

Same elsewhere in your code.

Reply:
Fixed: https://github.com/linkedin/TonY/pull/62/commits/f84caffeeea11f73f488a6d87502659e5c22c5b0